File: scripts/extract-land-points.py
"""Extract LAND_POINTS from agent transcript into lib/globe-land-points.json."""
import json
import re
from pathlib import Path

TRANSCRIPT = Path(
    r"C:\Users\umer.imran\.cursor\projects\d-figma\agent-transcripts"
    r"\166b00bf-64e6-491b-802e-071161d3e568"
    r"\166b00bf-64e6-491b-802e-071161d3e568.jsonl"
)
OUT = Path(__file__).resolve().parent.parent / "public" / "globe-land-points.json"


def extract_array(text: str) -> str:
    marker = "const LAND_POINTS = "
    start = text.rfind(marker)
    if start == -1:
        raise ValueError("LAND_POINTS not found")
    i = start + len(marker)
    if text[i] != "[":
        raise ValueError("Expected [ after LAND_POINTS =")
    depth = 0
    for j in range(i, len(text)):
        ch = text[j]
        if ch == "[":
            depth += 1
        elif ch == "]":
            depth -= 1
            if depth == 0:
                return text[i : j + 1]
    raise ValueError("Unclosed LAND_POINTS array")


def main() -> None:
    html = ""
    for line in TRANSCRIPT.read_text(encoding="utf-8").splitlines():
        if "const LAND_POINTS =" in line and '"role":"user"' in line:
            obj = json.loads(line)
            html = obj["message"]["content"][0]["text"]
    if not html:
        raise SystemExit("No user message with LAND_POINTS found")

    raw = extract_array(html)
    points = json.loads(raw)
    OUT.parent.mkdir(parents=True, exist_ok=True)
    OUT.write_text(json.dumps(points), encoding="utf-8")
    print(f"Wrote {len(points)} points to {OUT}")


if __name__ == "__main__":
    main()

File: scripts/extract-land-points.mjs
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const transcriptDir = path.join(
  process.env.USERPROFILE || process.env.HOME || "",
  ".cursor/projects/d-figma/agent-transcripts"
);

function extractArray(text) {
  const marker = "const LAND_POINTS = ";
  const start = text.lastIndexOf(marker);
  if (start === -1) throw new Error("LAND_POINTS not found");
  let i = start + marker.length;
  if (text[i] !== "[") throw new Error("Expected [ after LAND_POINTS =");
  let depth = 0;
  for (let j = i; j < text.length; j++) {
    const ch = text[j];
    if (ch === "[") depth++;
    else if (ch === "]") {
      depth--;
      if (depth === 0) return text.slice(i, j + 1);
    }
  }
  throw new Error("Unclosed LAND_POINTS array");
}

function findHtmlInTranscripts() {
  if (!fs.existsSync(transcriptDir)) return "";
  const files = fs.readdirSync(transcriptDir, { recursive: true });
  for (const rel of files) {
    if (typeof rel !== "string" || !rel.endsWith(".jsonl")) continue;
    const file = path.join(transcriptDir, rel);
    const stat = fs.statSync(file);
    if (!stat.isFile() || stat.size < 100_000) continue;
    for (const line of fs.readFileSync(file, "utf8").split(/\r?\n/)) {
      if (line.includes("const LAND_POINTS =")) return line;
    }
  }
  return "";
}

const line = findHtmlInTranscripts();
if (!line) {
  console.error("No transcript line containing LAND_POINTS found.");
  process.exit(1);
}

const obj = JSON.parse(line);
const html = obj.message?.content?.[0]?.text ?? "";
const raw = extractArray(html);
const points = JSON.parse(raw);

const outDir = path.join(root, "public");
fs.mkdirSync(outDir, { recursive: true });
const outFile = path.join(outDir, "globe-land-points.json");
fs.writeFileSync(outFile, JSON.stringify(points));
console.log(`Wrote ${points.length} points to ${outFile}`);
